content/docs/command-reference/exp/run.md

```diff
@@ -84,13 +84,23 @@ steps as `make_checkpoint()` — please refer to its reference for details.
 You can now use `dvc exp run` to begin the experiment. If the process gets
 interrupted (e.g. with `[Ctrl] C` or by an error), all the checkpoints so far
 will be preserved. When a run finishes normally, a final checkpoint will be
-added (if needed) to wrap up the experiment.
+added (if needed) to wrap up the experiment. Following uses of `dvc exp run`
+will continue from this point (using the latest cached versions of all outputs).
 
-Following uses of `dvc exp run` will continue from this point (using the latest
-cached versions of all outputs). You can add a `--rev` to continue from a
-previous checkpoint instead (list them with `dvc exp show`). Or use `--reset` to
-start over (discards previous checkpoints and deletes `checkpoint` outputs, like
-the first `dvc exp run`) — useful for re-training ML models, for example.
+To continue from a previous checkpoint (list them with `dvc exp show`), when
+running experiments in your workspace, you must first use `dvc exp apply` to
+apply the checkpoint from which you want to resume before using `dvc exp run`.
+When running experiments outside of your workspace via `--queue/--run-all` or
+`--temp`, you can use `--rev` to directly specify the checkpoint from which you
+want to resume.
+
+⚠️ Note that `--rev` is disallowed when running in your workspace, since using
+`--rev` in this way could potentially lead to discarding and overwriting changes
+in your workspace.
```

Comment:
```suggestion
⚠️ Note that `--rev` is disallowed when running in your workspace in order to
prevent uncommitted changes potentially being discarded/overwritten.
```

is "uncommitted" an acceptable term to use according to DVC nomenclature @jorgeorpinel?

Reply:
> is "uncommitted" an acceptable term

I'd just say "changes". We want to simplify the language as much as possible.

But TBH I'm not sure we need this whole note (I know @dberenbaum requested some motivation but also it looks a bit like an implementation detail, and we avoid those in docs in general).

Reply:
Presumably if you try to use `--rev` incorrectly you'll get a warning message saying "please use `exp apply` instead"?

Reply:
AFAIK actually https://github.com/iterative/dvc/blob/54c0675fc664cd97394a1f789aaa385a5205a304/dvc/command/experiments.py#L517-L521 is this a bug? It only checks for `queue or tmp_dir` but not `run_all`

Reply:
Yes, I agree this whole note seems unnecessary (the paragraph above already specifies that `--rev` is for use outside the workspace).

@casperdcl I think `--run-all` will silently ignore `--rev`. It actually probably silently ignores lots of arguments actually now that I think about since it's kind of a unique flag that doesn't create an experiment but runs existing ones. Not sure what's best here, although the `--queue` and `--run-all` flags might ultimately get replaced by a `exp queue` command.

Reply:
> Should we just remove this note/paragraph?
agree this whole note seems unnecessary

☠️ it!

> actually https://github.com/iterative/dvc/blob/54c0675fc664cd97394a1f789aaa385a5205a304/dvc/command/experiments.py#L517-L521 is this a bug?

@casperdcl did you have a chance to try it locally? Always best to use the tool and double check during the docs process, I think. But if still needed, please tag **pmrowla** for implementation Qs.

Reply:
> --run-all will silently ignore --rev ... probably silently ignores lots of arguments...
it's kind of a unique flag that doesn't create an experiment but runs existing ones

@dberenbaum great Qs but I think that's a slightly different Q. Let me reopen that as a separate review...

Reply:
> Let me reopen that as a separate review

Actually @casperdcl already mentioned it separately in https://github.com/iterative/dvc.org/pull/2383#pullrequestreview-644051647. I couldn't find a good place in the code for that review anyway, seems like it's not in the scope of this PR? But

> the --queue and --run-all flags might ultimately get replaced by a exp queue command

Is there an issue to track that @dberenbaum ? Thanks 🙂

Reply:
https://github.com/iterative/dvc/issues/5615

Reply:
nuked.